File: Easy/746 - MinCostClimbingStairs/solution.ts
// Question 746. Min Cost Climbing Stairs (https://leetcode-cn.com/problems/min-cost-climbing-stairs/)

/**
 * 执行用时：92 ms, 在所有 TypeScript 提交中击败了 88.89% 的用户
 * 内存消耗：40.7 MB, 在所有 TypeScript 提交中击败了 44.44% 的用户
 *
 * @param {number[]} cost
 * @returns {number}
 */
function minCostClimbingStairs(cost: number[]): number {
  const N: number = cost.length;
  const dp: number[] = [0, 0];

  for (let i = 2; i <= N; i++) {
    dp[i] = Math.min(dp[i - 2] + cost[i - 2], dp[i - 1] + cost[i - 1]);
  }

  return dp[N];
}

File: Easy/746 - MinCostClimbingStairs/Solution.cs
// Question 746. Min Cost Climbing Stairs (https://leetcode-cn.com/problems/min-cost-climbing-stairs/)

/// <summary>
/// 执行用时：108 ms, 在所有 C# 提交中击败了 85.88% 的用户
/// 内存消耗：26 MB, 在所有 C# 提交中击败了 8.44% 的用户
/// </summary>
public class Solution 
{
    public int MinCostClimbingStairs(int[] cost) 
    {
       int N = cost.Count();
       int[] dp = new int[N + 1];
       dp[0] = 0;
       dp[1] = 0;
       
       for (int i = 2; i <= N; i++) 
       {
           dp[i] = Math.Min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2]);
       }

       return dp[N];
    }
}